sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -547,8 +547,7 @@ class Analyzer(
       case a: Aggregate if containsStar(a.aggregateExpressions) =>
         if (conf.groupByOrdinal && a.groupingExpressions.exists(IntegerIndex.unapply(_).nonEmpty)) {
           failAnalysis(
-            "Group by position: star is not allowed to use in the select list " +
-              "when using ordinals in group by")
+            "Star (*) is not allowed in select list when GROUP BY ordinal position is used")
```

Comment:
See the original discussion: https://github.com/apache/spark/pull/10731#issuecomment-198034957